provisioning/device/src/transport_state_machine.ts

```diff
@@ -0,0 +1,239 @@
+                case 'assigned': {
+                  /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_014: [ If `TransportHandlers.registrationRequest` succeeds with status==Assigned, it shall emit an 'operationStatus' event and call `callback` with null, the response body, and the protocol-specific result. ] */
+                  /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_020: [ If `TransportHandlers.queryOperationStatus` succeeds with status==Assigned, `register` shall emit an 'operationStatus' event and complete and pass the body of the response and the protocol-spefic result to the `callback`. ] */
+                  this.emit('operationStatus', responseBody);
+                  this._fsm.transition('connected', callback, null, responseBody, result);
+                  break;
+                }
+                case 'assigning': {
+                  /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_015: [ If `TransportHandlers.registrationRequest` succeeds with status==Assigning, it shall emit an 'operationStatus' event and begin polling for operation status requests. ] */
+                  /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_021: [ If `TransportHandlers.queryOperationStatus` succeeds with status==Assigning, `register` shall emit an 'operationStatus' event and begin polling for operation status requests. ] */
+                  this.emit('operationStatus', responseBody);
+                  this._fsm.transition('waiting_to_poll', callback, registrationId, responseBody.operationId, pollingInterval);
+                  break;
+                }
+                default: {
+                  /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_016: [ If `TransportHandlers.registrationRequest` succeeds returns with an unknown status, `register` shall fail with a `SyntaxError` and pass the response body and the protocol-specific result to the `callback`. ] */
+                  /* Codes_SRS_NODE_PROVISIONING_TRANSPORT_STATE_MACHINE_18_022: [ If `TransportHandlers.queryOperationStatus` succeeds with an unknown status, `register` shall fail with a `SyntaxError` and pass the response body and the protocol-specific result to the `callback`. ] */
+                  let err = new SyntaxError('status is ' + responseBody.status);
+                  (err as any).result = result;
+                  (err as any).responseBody = responseBody;
+                  this._fsm.transition('connected', callback, err);
+                  break;
+                }
+              }
+            } else {  // err == null && responseBody == null
+              let err = this._transport.getErrorResult(result);
+              (err as any).result = result;
+              (err as any).responseBody = responseBody;
+              this._fsm.transition('connected', callback, err, responseBody, result);
+            }
+          },
+          responseReceived: () => { throw new Error(); },
```

Comment:
>new Error() [](start = 42, length = 11)

I imagine we're considering it's a bug to receive another response if we're already analyzing one considering the timer hasn't been restarted yet?
in that case, I'm ok with throwing but i think we should make the error explicit so that customers can tell us "hey, i ran into that"

Reply:
This shouldn't happen any more.  I've removed the reseponseReceived action and handle the cancel case directly in the callback from the transport.

---
In reply to: [147225350](https://github.com/Azure/azure-iot-sdk-node/pull/115#discussion_r147225350) [](ancestors = 147225350)